src/app/common/services/swapy-protocol.service.ts

```diff
@@ -48,6 +60,11 @@ export class SwapyProtocolService {
       return origCall.apply(this, arguments);
     };
   }
+  private storeTransactionHash(addresses: string[], hash: string) {
+    addresses.forEach(address => {
+      this.storageService.setItem(address, hash);
```

Comment:
why all the addresses point to the same hash?

Reply:
Let's suppose a user puts 3 assets on sale at the same time. With the new UI (and protocol) the user just needs to send one transaction via MetaMask. The contract internally will do the call for the 3 assets internally, but I just need the hash of the main transaction to see its status later.